Meshing/Feature-uniform-refining

miguelmaso opened the pull request:
A small change: using ModelPart*

miguelmaso committed:
uning ModelPart*

loumalouomega commented on applications/MeshingApplication/custom_utilities/uniform_refine_utility.h:
```diff
@@ -218,8 +218,8 @@ class UniformRefineUtility
     std::unordered_map<IndexType,int> mElemColorMap;
-    std::unordered_map<int,std::vector<std::string>> mColors;  /// Where the sub model parts IDs are stored
-    std::map<std::pair<int,int>,int> mIntersections;           /// Where the colors intersection are stored
+    std::unordered_map<int,std::vector<ModelPart*>> mColorsPointers; /// Where the sub model parts are stored
```
Be careful with the raw pointers!!!!, the raw pointers are evil (potential memory leak)

loumalouomega replied:
I recommend to use smart pointers if possible

miguelmaso replied:
I want to use ModelPart::Pointer but nowadays I think it is not possible...

loumalouomega replied:
Let me check...

loumalouomega reviewed (dismissed):
Please check the raw pointers for a future modification

loumalouomega committed:
Using smart pointers instead of raw pointers

Using ModelPart::Pointer p_main_model_part(&ThisModelPart); for the main model part

loumalouomega committed:
Revert "Using smart pointers instead of raw pointers"

This reverts commit 519ed5eb6da7c6dea81396c970e97bce7668b7cd.

loumalouomega commented:
@miguelmaso Can you check if this works in your machine, in mine works, but I have my reserves creating a smart pointer from the raw pointer (at the end is the same problem):

       ModelPart::Pointer p_main_model_part(&ThisModelPart);

https://stackoverflow.com/questions/4665266/creating-shared-ptr-from-raw-pointer

loumalouomega commented:
If doesn't work better to revert this one and learn to life in this wicked world with raw pointers

miguelmaso commented:
This is throwing a memory error. In addition, that change is breaking mmg process. I have locally reverted the changes and tried with another implementation, but I'm getting the same memory error...

loumalouomega commented:
Let's use just what you did

RiccardoRossi commented:
guys, YOU CAN NOT create a smart pointer from an existing pointer...

RiccardoRossi commented on kratos/tests/utilities/test_sub_model_parts_list_utility.cpp:
```diff
@@ -265,8 +265,8 @@ namespace Kratos
                         for (auto sub_model_part_name : colors[key]) {
-                            ModelPart& r_sub_model_part = SubModelPartsListUtility::GetRecursiveSubModelPart(second_model_part, sub_model_part_name);
-                            r_sub_model_part.AddNode(p_node);
+                            ModelPart::Pointer r_sub_model_part = SubModelPartsListUtility::GetRecursiveSubModelPart(second_model_part, sub_model_part_name);
```
why do you need a Pointer anyway?

RiccardoRossi commented on kratos/tests/utilities/test_sub_model_parts_list_utility.cpp:
```diff
@@ -348,7 +348,7 @@ namespace Kratos
             {
                 // Check both colors have the same size
-                std::vector<ModelPart*> p_color = colors_pointers[color.first];
+                std::vector<ModelPart::Pointer> p_color = colors_pointers[color.first];
```
why do you need to adquire ownership?

loumalouomega commented:
We know, I ask to @miguelmaso to revert it

loumalouomega reviewed (approved):
After reverting re-approving

miguelmaso commented:
I will use a different implementation
